src/golang/lib/engine/databricks_executor.go

```diff
@@ -0,0 +1,132 @@
+package engine
+
+import (
+	"context"
+	"time"
+
+	"github.com/aqueducthq/aqueduct/lib/collections/shared"
+	"github.com/aqueducthq/aqueduct/lib/job"
+	dag_utils "github.com/aqueducthq/aqueduct/lib/workflow/dag"
+	"github.com/aqueducthq/aqueduct/lib/workflow/operator"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"github.com/dropbox/godropbox/errors"
+)
+
+func ExecuteDatabricks(
+	ctx context.Context,
+	workflowDag dag_utils.WorkflowDag,
+	workflowName string,
+	workflowRunMetadata *WorkflowRunMetadata,
+	timeConfig *AqueductTimeConfig,
+	opExecMode operator.ExecutionMode,
+	databricksJobManager *job.DatabricksJobManager,
+) error {
+	inProgressOps := workflowRunMetadata.InProgressOps
+	completedOps := workflowRunMetadata.CompletedOps
+	dag := workflowDag
+
+	// Convert the operators into tasks
+	taskList, err := CreateTaskList(ctx, dag, databricksJobManager)
+	if err != nil {
+		return errors.Wrap(err, "Unable to convert operators to Databricks tasks.")
+	}
+
+	// Launch the workflow job with all tasks
+	_, err = databricksJobManager.LaunchMutlipleTaskJob(ctx, workflowName, taskList)
+	if err != nil {
+		return errors.Wrap(err, "Unable to launch workflow job on Databricks.")
+	}
+
+	for _, op := range dag.Operators() {
+		inProgressOps[op.ID()] = op
+	}
+
+	if len(inProgressOps) == 0 {
+		return errors.Newf("No initial operators to schedule.")
+	}
+
+	// Wait a little bit for all active operators to finish before exiting on failure.
+	defer waitForInProgressOperators(ctx, inProgressOps, timeConfig.OperatorPollInterval, timeConfig.CleanupTimeout)
+
+	start := time.Now()
+
+	for len(inProgressOps) > 0 {
```

Comment:
So, is it true that we don't need to do the checking of whether or not an operator is ready to run, because we provided a task dependency list when creating the databricks task?

Reply:
Yep, with aqueduct orchestration we would check that the operators dependencies had completed before calling op.Launch(). Once we submit the workflow job to Databricks it handles the dependency graph and launching jobs in the correct order.